### DISCUSSION TITLE
When running eslint with --config, why does it take the default config file and MERGE it?

### DISCUSSION THREAD

**Discussion by Industrial** [eslint/eslint#14131] [General]
Hi.

I have created a configuration file `.eslint-yml.json`:

```json
{
  "extends": [
    "plugin:yml/standard"
  ],
  "rules": {}
}
```

I did this because I am using `"parser": "@typescript-eslint/parser"` and this yml plugin will not work with it.

When I call `eslint -c .eslint-yml.json` when I have a `.eslint.json` file, it will MERGE the two files and cause an error.

Why does it look at the `.eslint.json` file at all?

**Comment by mdjermanovic** [comment]
Hi @Industrial!

This is the expected behavior for the `-c` option, as [documented](https://eslint.org/docs/user-guide/command-line-interface#-c-config):

> If .eslintrc.* and/or package.json files are also used for configuration (i.e., --no-eslintrc was not specified), the configurations will be merged. Options from this configuration file have precedence over the options from .eslintrc.* and package.json files.

You can additionally use the [--no-eslintrc](https://eslint.org/docs/user-guide/command-line-interface#-no-eslintrc) option to disable `.eslintrc.*` configs:

```
eslint -c .eslint-yml.json --no-eslintrc
```

> Why does it look at the .eslint.json file at all?

There are use cases where the `-c` config is used to just modify the configuration from the `.eslintrc.*` configs, so this option doesn't disable `.eslintrc.*` configs by default since that can be controlled with the `--no-eslintrc` option.

**Reply by Industrial** [reply]
Thank you for this. Right now I am moving the `.eslintrc.json` file before running and moving it back after so I'll definitely use this solution :)
